### PR TITLE
Configurable LogEntry creation with logback appender

### PR DESCRIPTION
**Pull request by favila**
This adds a knob to the Logback Appender that allows the user to customize how a `LogEntry` is built from an `ILoggingEvent`. My motivation was I wanted to log the logger name, but the `LoggingEnhancer` interface does not have access to the original log event.

The implementation strategy is similar to `LoggingEnhancer`. I created an interface `ILoggingEventTransformer` which accepts an `ILoggingEvent` and must return a `LogEntry.Builder` which may be passed through Enhancers. `logback.xml` can supply the classname of (only one!) logging event transformer via `<transformer>`.

I also refactored the existing `LogEntry`-building code (mostly the `logEntryFor()` private method) out of `LoggingAppender` into `StringPayloadTransformer` and a static class of utility functions.

This also makes it easy to use payloads other than strings (e.g. jsonPayload) for structured logging, which should satisfy #2669. I include an example of this approach in `TestJsonPayloadTransformer`.

### CODE REVIEW

File: google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingEventTransformerUtils.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.StackTraceElementProxy;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Severity;
+
+/**
+ * Collection of constants and static methods useful for writing LoggingEvent transformers
+ * {@link ILoggingEventTransformer}
+ */
+public final class LoggingEventTransformerUtils {
```
[the remark this answers is not in the document]
Comment (favila) [reply google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingEventTransformerUtils.java]: I added a private constructor to the first commit.

File: google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingEventTransformerUtils.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  private LoggingEventTransformerUtils(){}
+
+  /**
+   * Label to annotate log level names (e.g. "DEBUG") from {@link ILoggingEvent#getLevel()}.
+   */
+  public static final String LEVEL_NAME_KEY = "levelName";
+  /**
+   * Label to annotate log level numbers (e.g. "10000") from {@link ILoggingEvent#getLevel()}}.
+   */
+  public static final String LEVEL_VALUE_KEY = "levelValue";
```
[the remark this answers is not in the document]
Comment (favila) [reply google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingEventTransformerUtils.java]: Moved the constructor down

**Comment by pongad** [comment]
@jabubake Do you want to take a look?

**Comment by skjolber** [comment]
So would it be possible to add a test-case for JSON output using [logback-logstash-encoder](https://github.com/logstash/logstash-logback-encoder)? 

Or perhaps adding a new google-cloud-logging-logback-logstash module would be appropriate?

**Comment by sduskis** [comment]
@favila, I'm going to close this PR due to staleness.  Please feel free to resolve the conflicts and reopen.

**Pull request closed without merging**